### ISSUE TITLE
For slow progress, the estimated time left seems to go in the wrong direction

### ISSUE THREAD

**Issue by sboukortt** [cheggaaa/pb#112]
For a program with items that each take, say, a few seconds to complete, the estimated time left correctly decreases the instant a new item is done, but in between items, it increases. (Basically, the library will print something like: “0:30… 0:31… 0:32… [new item] 0:27… 0:28… 0:29… [new item] 0:24…”)

From reading the source code, I have the impression that it is because it does not keep track of when the last item was completed. Consequently, the time spent since then counts toward the average time per item, instead of being subtracted.

What also seems to support that hypothesis is that the increase gets slower as more items are completed.

To summarize, I think that the correct formula would be:

    average_time_per_item = time_spent_before_last_item_completed / number_of_items_completed
    time_left = average_time_per_item × number_of_items_left − time_spent_since_last_completed

But instead, the current code is equivalent to this:

    total_time = time_spent_before_last_item_completed + time_spent_since_last_item_completed
    average_time_per_item = total_time / number_of_items_completed
    time_left = average_time_per_item × number_of_items_left

**Comment by cheggaaa** [comment]
Thank you for the report.
I added changes that you advised.
Later it will be fixed in [v2](https://github.com/cheggaaa/pb/tree/v2) (there is a bit more difficult because of EWMA)

**Comment by sboukortt** [comment]
That was quick, thank you!

**Comment by viru** [comment]
Hi,
I have a different problem. I'm uploading a lot of big files (350GB total, 20-30MB each) and it seems that ETA is going down and then back to the previous point. You can see this behaviour on this GIF: https://giphy.com/gifs/l1J9qjphwdoTfBmOA/
